src/aura/CRLP_FilterGroup/CRLP_FilterGroupHelper.js

```diff
@@ -153,26 +163,129 @@
+                    var state = response.getState();
+                    if (state === "SUCCESS") {
+                        // Response will be a serialized deployResult wrapper class
+                        var deployResult = JSON.parse(response.getReturnValue());
+                        console.log('deployResult=' + deployResult);
+                        // if there is a record id response
+                        if (deployResult && deployResult.completed === true && deployResult.filterGroupItem) {
+                            window.clearTimeout(poller);
+                            helper.toggleSpinner(cmp, false);
+                            helper.showToast(cmp, 'success', cmp.get("v.labels.filtersSaveProgress"), cmp.get("v.labels.filtersSaveSuccess"));
+
+                            var model = helper.restructureResponse(deployResult.filterGroupItem);
+
+                            // Save the inserted/updated record id
+                            cmp.set("v.activeFilterGroup", model.filterGroup);
+                            cmp.set("v.activeFilterGroupId", model.filterGroup.recordId);
+
+                            // for a new record, copy the activeFilterGroup map to the cachedFilterGroup map
+                            if (cmp.get("v.cachedFilterGroup") && cmp.get("v.cachedFilterGroup.recordName")) {
+                                var activeFilterGroup = cmp.get("v.activeFilterGroup");
+                                var cachedFilterGroup = cmp.get("v.cachedFilterGroup");
+                                for (var key in activeFilterGroup) {
+                                    if (activeFilterGroup.hasOwnProperty(key)) {
+                                        cachedFilterGroup[key] = activeFilterGroup[key];
+                                    }
+                                }
+                                cmp.set("v.cachedFilterGroup", cachedFilterGroup);
+                            }
+
+                            var filterRuleList = helper.restructureResponse(model.filterRuleList);
+                            var filterRuleListCached = helper.restructureResponse(model.filterRuleList);
+
+                            cmp.set("v.filterRuleList", filterRuleList);
+                            cmp.set("v.cachedFilterRuleList", filterRuleListCached);
+                            var rollupItems = cmp.get("v.rollupItems");
+
+                            // need to sent back an object with the following properties
+                            var filterGroupTableItem = {};
+                            filterGroupTableItem.label = model.filterGroup.label;
+                            filterGroupTableItem.description = model.filterGroup.description;
+                            filterGroupTableItem.name = model.filterGroup.recordName;
+                            filterGroupTableItem.recordId = model.filterGroup.recordId;
+                            if (filterRuleList) {
```

Comment:
Couldn't you send back the deployResult.filterGroupItem like you did in rollups, just with a few extra properties attached to it? Or is it preferable to make a new object here?

Reply:
The issue here was that the values needed in the RollupsContainer were not quite the same as the straight filterGroupItem wrapper.